apps/etl/extraction/sources/pdc/extract.py

```diff
@@ -63,234 +76,116 @@ class PdcHazardInputMetadata(pydantic.BaseModel):
-        headers = {"Authorization": f"Bearer {etl_config.PDC_SENTRY_AUTHORIZATION_KEY}"}
-        return (requests.get(url, headers=headers).json(), url)
-
-    @classmethod
-    def store_pdc_data(
-        cls,
-        response,
-        source: ExtractionData.Source,
-        url: str,
-        validate_source_func: typing.Any | None = None,
-        # FIXME: intance_id is always None. This is a bug.
-        instance_id: typing.Any | None = None,
-        parent: ExtractionData | None = None,
-        hazard_type: HazardType | None = None,
-        metadata: typing.Any | None = None,
-    ):
-        file_name = f"{instance_id}pdc.json"
-        data = json.dumps(response).encode("utf-8")
-        instance = cls._create_extraction_instance(
-            url="",
-            source=source,
-            parent_id=parent and parent.id,
-            status=ExtractionData.Status.SUCCESS,
-            hazard_type=hazard_type,
-            metadata=metadata,
+    def handle_type_hazard(self):
+        self._extraction_fetch_url(
+            self.extraction_metadata.url,
+            data=json.dumps(self.extraction_metadata.extraction_args),
+            headers={
+                "Authorization": f"Bearer {etl_config.PDC_SENTRY_AUTHORIZATION_KEY}",
+                "Content-Type": "application/json",
+            },
+            method="post",
         )
-
-        content_file = ContentFile(data)
-        content_file.name = file_name
-        instance.resp_data.save(content_file.name, content_file)
-
-        return instance
-
-    @classmethod
-    # FIXME: Write pydantic type for hazard
-    def process_hazard(cls, instance: ExtractionData, hazard: dict):
-        try:
-            if hazard["type_ID"] not in HAZARD_TYPE_MAP.keys():
-                logger.warning(
-                    "Skipping extraction because hazard is not supported",
-                    extra=log_extra(
-                        dict(
-                            extraction_id=instance.id,
-                            hazard_uuid=hazard["type_ID"],
-                        )
-                    ),
-                )
-                return
-
-            hazard_uuid = hazard["uuid"]
-            hazard_type_id = hazard["type_ID"]
-
-            geo_json_file, geo_json_url = cls.fetch_geo_json(hazard_uuid)
-            geo_json_data = cls.store_pdc_data(
-                response=geo_json_file,
-                source=ExtractionData.Source.PDC,
-                url=geo_json_url,
-                validate_source_func=None,
-                parent=instance,
-                hazard_type=HAZARD_TYPE_MAP.get(hazard_type_id),
-                metadata={
-                    "input": PdcPolygonInputMetadata(
-                        hazard_uuid=hazard_uuid,
-                        hazard_type_id=hazard_type_id,
-                    ).model_dump(),
-                },
+        response_data = json.loads(self.extraction_object.resp_data.read())
```

Comment:
Tip: Make sure this works if used
```suggestion
        response_data =self.extraction_object.resp_data.json()
```